src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedItems: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const groupBy = String(item[key]);
+
+    if (groupBy in groupedItems) {
```

Comment:
```suggestion
    if ({}.hasOwnProperty.call(groupedItems, groupBy) {
```
The proper way to check this